### PR TITLE
[SPARK-42011][CONNECT][PYTHON] Implement DataFrameReader.csv

### PR DESCRIPTION
**Pull request by techaddict**
### What changes were proposed in this pull request?
This PR implements `DataFrameReader.csv` alias in Spark Connect.

### Why are the changes needed?
For API feature parity.

### Does this PR introduce any user-facing change?
This PR adds a user-facing API but Spark Connect has not been released yet.

### How was this patch tested?
Unittest was added.

### CODE REVIEW

File: python/pyspark/sql/connect/readwriter.py [hunk 1]
```diff
@@ -514,6 +586,7 @@ def _test() -> None:
         del pyspark.sql.connect.readwriter.DataFrameReader.load.__doc__
         del pyspark.sql.connect.readwriter.DataFrameReader.option.__doc__
         del pyspark.sql.connect.readwriter.DataFrameReader.text.__doc__
+        del pyspark.sql.connect.readwriter.DataFrameReader.csv.__doc__
```
Comment (HyukjinKwon) [inline python/pyspark/sql/connect/readwriter.py:589]: Hm, does the test fail?

Comment (techaddict) [reply python/pyspark/sql/connect/readwriter.py]: It passed locally, was working against an older master branch.

**Comment by techaddict** [comment]
Updated the PR, tests will start passing once #39553 is merged, as its using `path: PathOrPaths`

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
That PR is merged. Mind retriggering the test?

**Review by zhengruifeng — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**